### DISCUSSION TITLE
How to parse a date from a string with the last_two year representation enabled.

### DISCUSSION THREAD

**Discussion by thepackett** [time-rs/time#570] [Question] [answered]
As per the title, I am trying to parse a date string of the format:
[month padding:none]/[day]/[year repr:last_two base:calendar]
i.e. 4/09/23, 11/27/09, etc.

Using Date::parse(). However I am getting the error that the "'Parsed' struct did not include enough information to construct the type". I suspect this is related to the last_two representation of the year being ambiguous to the parser, however I don't know how I am supposed to use the last_two year representation if that is the case.

How is the last_two year representation intended to be used?

On a side note, how does the padding field interact with the last_two representation? The documentation states that the padding has a width of four, is this reduced to two when the last_two representation is used?

**Comment by jhpratt** [comment, marked as answer]
You're correct that the parser ignores the "last two" representation when parsing due to the ambiguity. Right now, the only way to handle this is to go through the `time::parsing::Parsed` struct. For example:

```rust
use time::parsing::Parsed;
use time::macros::*;

const FORMAT: &[FormatItem<'_>] = format_description!("[month padding:none]/[day]/[year repr:last_two]");

// replace this with the actual input
let input = "4/09/23";

// parse the items from the input
let mut parsed = Parsed::new();
parsed.parse_items(input, FORMAT)?;

// set the year however you want
let year = parsed.year_last_two()? + 2000;
parsed.set_year(year)?;

// convert into the final type
let date = Date::try_from(parsed)?;
```

I've written this off-hand, so minor modifications will likely be necessary to make it compile.

As an aside, you don't need to specify `base:calendar` for year; it's the default.

**Reply by thepackett** [reply]
I just tried it out and confirmed that your code is correct as is after a couple of type casts, and working as intended. I'll be marking this as the correct answer, thanks for the help!
